pandas/core/indexes/datetimes.py

```diff
@@ -535,8 +554,6 @@ def _fast_union(self, other):
 
     def _wrap_setop_result(self, other, result):
         name = get_op_result_name(self, other)
-        if not timezones.tz_compare(self.tz, other.tz):
-            raise ValueError('Passed item and index have different timezone')
```

Comment:
this check is just redundant?

Reply:
It's not tested... so I hope so. I'll see if I can construct a case that hits it, but I suspect that we've converted to object-dtype in the setop if the tzs don't match.

Reply:
I wasn't able to make an example that hits this.